updatecli/scripts/check-jdk.sh

```diff
@@ -33,8 +33,10 @@ function get_jdk_download_url() {
       echo "https://github.com/adoptium/temurin19-binaries/releases/download/jdk-${jdk_version}/OpenJDK19U-jdk_${platform}_hotspot_${jdk_version//+/_}";
       return 0;;
     21*)
-      ## JDK21 URLs have an underscore ('_') instead of a plus ('+') in their archive names
-      echo "https://github.com/adoptium/temurin21-binaries/releases/download/jdk-${jdk_version_urlencoded}/OpenJDK21U-jdk_${platform}_hotspot_ea_21-0-${jdk_build_number//-ea-beta/}";
+      # JDK version (21+35)
+      ##    https://github.com/adoptium/temurin21-binaries/releases/download/jdk-21%2B35/OpenJDK21U-jdk_aarch64_linux_hotspot_21_35.tar.gz
```

Comment:
```suggestion
      ## JDK21 URLs have an underscore ('_') instead of a plus ('+') in their archive names
```

Let's stay homogeneous with the other JDKs